### PR TITLE
ISPN-9601 JMX cluster switching

### PR DESCRIPTION
**Pull request by tristantarrant**
https://issues.jboss.org/browse/ISPN-9601
https://issues.jboss.org/browse/ISPN-9602

Introduces thread name prefix/suffix for the client async pool.

Let's see if https://issues.jboss.org/browse/ISPN-9064 is also resolved

### CODE REVIEW

**Comment by tristantarrant** [comment]
The *fix* for ISPN-9064 is to use 127.0.0.1 instead of localhost in the client configuration.
When switching sites, the pools with the failed servers are closed and new pools for the added servers are created. Unfortunately, in the test, sometimes the subsequent PUT operation would still be executed with a channel from the old pool: the request would be processed by the server, but the client would ignore the response because the channel was invalidated in flight. This causes the client retry logic to invoke the operation a second time. @rvansa while the test should be now fixed, I believe the underlying issue should warrant its own Jira.

**Comment by danberindei** [comment]
@tristantarrant how exactly does 127.0.0.1 help? Otherwise LGTM.

**Comment by tristantarrant** [comment]
the client configuration points to localhost:port for the backup site, but when switching, it performs a ping and receives the topology for the backup cluster, which includes 127.0.0.1:port. This causes the "localhost:port" pools to be invalidated and "127.0.0.1:port" pools to be created instead. For some reason the PUT after the switch is being created with a "localhost:port" connection, which gets invalidated soon after sending to the server and then sent again via a connection from the "127.0.0.1:port". The server actually executes 2 PUTs.

**Comment by danberindei** [comment]
Could you put this explanation in a comment in the test @tristantarrant? I'm definitely not going to remember it :)

**Comment by tristantarrant** [comment]
Added it

**Comment by tristantarrant** [comment]
1 unrelated failure

File: client/hotrod-client/src/test/java/org/infinispan/client/hotrod/HitsAwareCacheManagersTest.java [hunk 1]
```diff
@@ -188,26 +190,33 @@ protected void destroy() {
...
+      private final AtomicInteger localSiteInvocationCount = new AtomicInteger(0);
+      private final AtomicInteger backupSiteInvocationCount = new AtomicInteger(0);
 
       @Override
-      public Object visitCommand(InvocationContext ctx, VisitableCommand command) throws Throwable {
+      public Object visitCommand(InvocationContext ctx, VisitableCommand command) {
          if (ctx.isOriginLocal()) {
-            int count = invocationCount.incrementAndGet();
-            log.infof("Hit %d for %s", count, command);
+            if ((command instanceof AbstractDataCommand) && ((AbstractDataCommand)command).getFlags().contains(Flag.SKIP_XSITE_BACKUP)) {
```
Comment (danberindei) [inline client/hotrod-client/src/test/java/org/infinispan/client/hotrod/HitsAwareCacheManagersTest.java]:
```suggestion
            if ((command instanceof AbstractDataCommand) && ((AbstractDataCommand)command).hasAnyFlag(FlagBitSets.SKIP_XSITE_BACKUP)) {
```
(needs an import as well)

**Comment by tristantarrant** [comment]
Applied your suggestion @danberindei

**Comment by danberindei** [comment]
Integrated, thanks Tristan!

**Pull request merged**